packages/engine/src/worker/runner/javascript/mod.rs

```diff
@@ -1257,26 +1665,30 @@ impl<'m> RunnerImpl<'m> {
             state_group_start_indices,
         } = ctx_batch_sync;
 
-        let args = mv8::Values::from_vec(vec![
-            sim_id_to_js(mv8, sim_run_id),
-            batch_to_js(
-                mv8,
-                context_batch.segment().memory(),
-                context_batch.segment().persisted_metaversion(),
-            )?,
-            idxs_to_js(mv8, &state_group_start_indices)?,
-            current_step_to_js(mv8, current_step),
-        ]);
-        let _: mv8::Value<'_> = self
-            .embedded
+        let js_sim_id = sim_id_to_js(scope, sim_run_id);
+        let js_batch_id = batch_to_js(
+            scope,
+            context_batch.segment().memory(),
+            context_batch.segment().persisted_metaversion(),
+        )?;
+        let js_idxs = new_js_array_from_usizes(scope, &state_group_start_indices)?;
+        let js_current_step = current_step_to_js(scope, current_step);
+        self.embedded
             .ctx_batch_sync
-            .call_method(self.this.clone(), args)?;
+            .call(scope, self.this, &[
+                js_sim_id,
+                js_batch_id,
+                js_idxs,
+                js_current_step,
+            ])
+            .ok_or_else(|| "Could not run ctx_batch_sync Function".to_string())?;
```

Comment:
Same as above as @Alfred-Mountfield noticed.

Reply:
For this one the function is literally called "ctx_batch_sync" in the runner.js.
What do you want specified in the error?

Reply:
IMO the function in runner.js should be renamed as well, but let's keep it as it is, sorry for the noise.